### PR TITLE
Fix Type.GetHashCode for RuntimeTypes w/o MethodTable

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

File: src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Type/TypePropertyTests.cs [hunk 1]
```diff
@@ -377,6 +377,13 @@ public void IsVariableBoundArray_Get_ReturnsExpected()
         {
             Assert.Equal(IsVariableBoundArray, CreateType().IsVariableBoundArray);
         }
+
+        [Fact]
+        public void GetHashCode_ReturnsExpected()
+        {
+            Type type = CreateType();
+            Assert.Equal(type.GetHashCode(), type.GetHashCode());
```
Comment (jkotas) [inline src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Type/TypePropertyTests.cs:385]:
```suggestion
            Assert.Equal(type.GetHashCode(), type.GetHashCode() + 1);
```

**Comment by jkotas** [comment]
@MichalStrehovsky PTLA. It is fixing a minor bug introduced by the RuntimeType refactoring.

**Review by MichalStrehovsky — APPROVED** [review]
Thanks!

Do we want to add the test you had here before the force push?

**Pull request merged**